runners/runner-local/src/manager.rs

```diff
@@ -0,0 +1,215 @@
+use std::fs::{create_dir_all, File};
+
+use std::path::PathBuf;
+use std::process::Stdio;
+use std::{collections::HashMap, sync::Arc};
+use tokio::sync::Mutex;
+
+use tracing::{info, warn};
+
+use crate::error::{LocalRunnerError, LocalRunnerResult};
+
+use apibara_runner_common::runner::v1::Indexer;
+
+use error_stack::ResultExt;
+use tokio::process::Command;
+
+use crate::server::IndexerInfo;
+use crate::utils::{build_indexer_command, refresh_status};
+
+pub struct IndexerManager {
+    pub indexers: Arc<Mutex<HashMap<String, IndexerInfo>>>,
+}
+
+impl Default for IndexerManager {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl IndexerManager {
+    pub fn new() -> Self {
+        Self {
+            indexers: Arc::new(Mutex::new(HashMap::new())),
+        }
+    }
+
+    pub async fn create_indexer(
+        &self,
+        indexer_id: String,
+        indexer: Indexer,
+    ) -> LocalRunnerResult<()> {
+        let command = build_indexer_command(&indexer_id, &indexer)?;
+
+        let port = portpicker::pick_unused_port()
+            .ok_or(LocalRunnerError::Internal)
+            .attach_printable("Can't pick a port for the status server")?;
+
+        let status_server_address = format!("0.0.0.0:{port}");
+
+        let status_server_args = &[
+            "--status-server-address".to_string(),
+            status_server_address.clone(),
+        ];
+
+        let status_server_address = format!("http://{status_server_address}");
+
+        let cmd = format!(
+            "{} {} {}",
+            command.program,
+            command.args.join(" "),
+            status_server_args.join(" ")
+        );
+        info!("Starting indexer {} with command `{}`", indexer.name, cmd);
+
+        let logs_dir = "logs";
```

Comment:
Maybe the logs directorly should be a temporary folder? If the user wants logs in a specific directory, we can add an option for it.

Reply:
OK